interfaces/builtin/juju_client_observe.go

```diff
@@ -0,0 +1,46 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const jujuClientObserveSummary = `allows read access to juju client configuration`
+
+const jujuClientObserveBaseDeclarationSlots = `
+  juju-client-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const jujuClientObserveConnectedPlugAppArmor = `
+# Description: Can access juju client configuration
+owner @{HOME}/.local/share/juju/{,**} r,
```

Comment:
I think we ought to define SNAP_REAL_HOME or something similar so that it can be reliably set and not guessed from /home/$LOGNAME. Otherwise each of the interfaces that allow access to something in real home force developers to guess.

Reply:
@zyga - are you talking about exporting an environment variable? If so, I am not against that (though I would say (and have!) people should use `getent passwd $UID` instead of guessing anyway).

Reply:
@jdstrand yes, I was thinking about a variable. The variable is easier to get and perhaps easier to discover.